src/runtime/security/client_negotiation.cpp

```diff
@@ -172,16 +167,18 @@ void client_negotiation::on_challenge(const negotiation_response &challenge)
 void client_negotiation::select_mechanism(const std::string &mechanism)
 {
     _selected_mechanism = mechanism;
+    _status = negotiation_status::type::SASL_SELECT_MECHANISMS;
 
-    auto req = dsn::make_unique<negotiation_request>();
-    _status = req->status = negotiation_status::type::SASL_SELECT_MECHANISMS;
-    req->msg = mechanism;
-    send(std::move(req));
+    send(_status, std::move(mechanism));
```

Comment:
`mechanism` is a 'const &', is it movable? there is not any compile complains?

Reply:
I didn't see any compile complains. And I write a test program like this:
```
    std::string test = "123123";
    const std::string &ref_test = test;
    const std::string &&r_test = std::move(ref_test);
    std::cout << r_test << std::endl;
```
It works well.